src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+using Uno.Extensions.Navigation.UI.Controls;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal record ForceBindingsUpdateGenerationContext(
+	GeneratorExecutionContext Context,
+
+	// Core types
+	[ContextType("Uno.Extensions.Navigation.IForceBindingsUpdate")] INamedTypeSymbol ForceBindingsUpdateInterface,
+
+	// Attributes
+	[ContextType(typeof(ForceUpdateAttribute))] INamedTypeSymbol UpdateAttribute,
+
+	// General stuff types
+	[ContextType("Microsoft.UI.Xaml.Controls.Page")] INamedTypeSymbol Page)
+{
+	public bool IsGenerationNotDisable(ISymbol symbol)
+		=> IsGenerationEnabled(symbol) ?? true;
+
+	public bool? IsGenerationEnabled(ISymbol symbol)
+		=> symbol.FindAttributeValue<bool>(UpdateAttribute, nameof(ForceUpdateAttribute.IsEnabled), 0) is { isDefined: true } attribute
+			? attribute.value ?? true
+			: null;
+
+	private ImmutableDictionary<string, AdditionalText>? _XBindFiles;
+	private static Regex ClassRegEx = new Regex("x:Class=\"([\\w.]+)\"");
+	private static Regex xBindRegEx = new Regex("{x:Bind ");
+
+	private IDictionary<string, AdditionalText> XBindFiles
+	{
+		get
+		{
+			if (_XBindFiles is null)
+			{
+				var files = new Dictionary<string, AdditionalText>();
+				foreach (var file in Context.AdditionalFiles)
+				{
+					var txt = file.GetText()?.ToString();
+					if (txt is not null &&
+						xBindRegEx.IsMatch(txt))
+					{
+						var classNameMatch = ClassRegEx.Match(txt);
+						if (classNameMatch.Success &&
+							classNameMatch.Groups.Count > 1)
+						{
+							files[classNameMatch.Groups[1].Value] = file;
+						}
+					}
+				}
+				_XBindFiles = files.ToImmutableDictionary();
+			}
+			return _XBindFiles;
+		}
+	}
+
+	public bool ContainsXBind(INamedTypeSymbol symbol)
+	{
+		try
+		{
+			var name = symbol.ToDisplayString();
+			return XBindFiles.ContainsKey(name);
+		}
+		catch { }
```

Comment:
I can't see any exception the above code can throw. `ContainsKey` throws when the argument is null, but `name` shouldn't be null.

Reply:
removed catch